From f19e66547e665835ead26b6f8b00341488ab8ec5 Mon Sep 17 00:00:00 2001
From: Randy Dunlap <rdunlap@infradead.org>
Date: Sun, 8 Nov 2020 19:21:15 -0800
Subject: [PATCH 12/13] clk: pm_clock: provide stubs for
 pm_clk_runtime_suspend/_resume
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add stubs for pm_clk_runtime_suspend() and pm_clk_runtime_resume()
to fix build errors when CONFIG_PM and CONFIG_PM_CLK are not enabled.

Fixes these build errors:

../drivers/clk/qcom/camcc-sc7180.c: In function ‘cam_cc_sc7180_probe’:
../drivers/clk/qcom/camcc-sc7180.c:1672:8: error: implicit declaration of function ‘pm_clk_runtime_resume’; did you mean ‘pm_runtime_resume’? [-Werror=implicit-function-declaration]
  ret = pm_clk_runtime_resume(&pdev->dev);
        ^~~~~~~~~~~~~~~~~~~~~
../drivers/clk/qcom/camcc-sc7180.c:1681:3: error: implicit declaration of function ‘pm_clk_runtime_suspend’; did you mean ‘pm_runtime_suspend’? [-Werror=implicit-function-declaration]
   pm_clk_runtime_suspend(&pdev->dev);
   ^~~~~~~~~~~~~~~~~~~~~~

Fixes: 15d09e830bbc ("clk: qcom: camcc: Add camera clock controller driver for SC7180")
Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
Reviewed-by: Nathan Chancellor <natechancellor@gmail.com>
Cc: "Rafael J. Wysocki" <rjw@rjwysocki.net>
Cc: Len Brown <len.brown@intel.com>
Cc: Pavel Machek <pavel@ucw.cz>
Cc: linux-pm@vger.kernel.org
Cc: Michael Turquette <mturquette@baylibre.com>
Cc: Stephen Boyd <sboyd@kernel.org>
Cc: linux-clk@vger.kernel.org
Cc: Taniya Das <tdas@codeaurora.org>
Cc: linux-next@vger.kernel.org
Cc: Nathan Chancellor <natechancellor@gmail.com>
Cc: Andrew Morton <akpm@linux-foundation.org>
Link: https://lore.kernel.org/r/20201109032115.10610-1-rdunlap@infradead.org
Build-tested-by: Nathan Chancellor <natechancellor@gmail.com>
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 include/linux/pm_clock.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/linux/pm_clock.h b/include/linux/pm_clock.h
index 8ddc7860e131..a0c72fce4689 100644
--- a/include/linux/pm_clock.h
+++ b/include/linux/pm_clock.h
@@ -27,6 +27,14 @@ extern int pm_clk_runtime_resume(struct device *dev);
 	.runtime_resume = pm_clk_runtime_resume,
 #else
 #define USE_PM_CLK_RUNTIME_OPS
+static inline int pm_clk_runtime_suspend(struct device *dev)
+{
+	return 0;
+}
+static inline int pm_clk_runtime_resume(struct device *dev)
+{
+	return 0;
+}
 #endif
 
 #ifdef CONFIG_PM_CLK
-- 
2.29.2

